modules/goraclebackend/soracle.cc

```diff
@@ -359,10 +327,61 @@ class SOracleStatement: public SSqlStatement {
     return d_query;
   }
 
-  ~SOracleStatement() { 
+  ~SOracleStatement() {
+    releaseStatement();
+  }
+
+private:
+
+  void initStatement() {
+    if (d_query.size()==0) return;
+    if (d_prepared) return;
+
+    if (OCIHandleAlloc(d_ctx, (dvoid**)&d_err, OCI_HTYPE_ERROR, 0, NULL))
+      throw SSqlException("Cannot allocate statement error handle");
+
+    if (d_release_stmt) {
+      if (OCIStmtPrepare2(d_svcctx, &d_stmt, d_err, (text*)d_query.c_str(), d_query.size(),
+          NULL, 0, OCI_NTV_SYNTAX, OCI_DEFAULT) != OCI_SUCCESS) {
+        // failed.
+        throw SSqlException("Cannot prepare statement (1): " + d_query + string(": ") + OCIErrStr());
+      }
+      d_init = true;
+    } else d_init = false;
```

Comment:
please don't have a single line else branch with no braces after a multi-line if branch (for readability)